Build warnings with clang-6.

Clang 6 warns about mismatched enums:

external/protobuf_git/src/google/protobuf/compiler/cpp/cpp_message.cc:205:10: error: comparison of two values with different enumeration types in switch statement ('FieldD
escriptor::CppType' and 'google::protobuf::internal::WireFormatLite::CppType') [-Werror,-Wenum-compare-switch]
    case internal::WireFormatLite::CPPTYPE_BOOL:
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/protobuf_git/src/google/protobuf/compiler/cpp/cpp_message.cc:203:10: error: comparison of two values with different enumeration types in switch statement ('FieldD
escriptor::CppType' and 'google::protobuf::internal::WireFormatLite::CppType') [-Werror,-Wenum-compare-switch]
    case internal::WireFormatLite::CPPTYPE_DOUBLE:
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/protobuf_git/src/google/protobuf/compiler/cpp/cpp_message.cc:201:10: error: comparison of two values with different enumeration types in switch statement ('FieldD
escriptor::CppType' and 'google::protobuf::internal::WireFormatLite::CppType') [-Werror,-Wenum-compare-switch]
    case internal::WireFormatLite::CPPTYPE_FLOAT:
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/protobuf_git/src/google/protobuf/compiler/cpp/cpp_message.cc:199:10: error: comparison of two values with different enumeration types in switch statement ('FieldD
escriptor::CppType' and 'google::protobuf::internal::WireFormatLite::CppType') [-Werror,-Wenum-compare-switch]
    case internal::WireFormatLite::CPPTYPE_UINT64:
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/protobuf_git/src/google/protobuf/compiler/cpp/cpp_message.cc:197:10: error: comparison of two values with different enumeration types in switch statement ('FieldD
escriptor::CppType' and 'google::protobuf::internal::WireFormatLite::CppType') [-Werror,-Wenum-compare-switch]
    case internal::WireFormatLite::CPPTYPE_UINT32:
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/protobuf_git/src/google/protobuf/compiler/cpp/cpp_message.cc:195:10: error: comparison of two values with different enumeration types in switch statement ('FieldD
escriptor::CppType' and 'google::protobuf::internal::WireFormatLite::CppType') [-Werror,-Wenum-compare-switch]
    case internal::WireFormatLite::CPPTYPE_INT64:
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

--- a/src/google/protobuf/compiler/cpp/cpp_message.cc
+++ b/src/google/protobuf/compiler/cpp/cpp_message.cc
@@ -113,16 +113,16 @@ struct ExtensionRangeSorter {
 bool IsPOD(const FieldDescriptor* field) {
   if (field->is_repeated() || field->is_extension()) return false;
   switch (field->cpp_type()) {
-    case internal::WireFormatLite::CPPTYPE_ENUM:
-    case internal::WireFormatLite::CPPTYPE_INT32:
-    case internal::WireFormatLite::CPPTYPE_INT64:
-    case internal::WireFormatLite::CPPTYPE_UINT32:
-    case internal::WireFormatLite::CPPTYPE_UINT64:
-    case internal::WireFormatLite::CPPTYPE_FLOAT:
-    case internal::WireFormatLite::CPPTYPE_DOUBLE:
-    case internal::WireFormatLite::CPPTYPE_BOOL:
+    case FieldDescriptor::CPPTYPE_ENUM:
+    case FieldDescriptor::CPPTYPE_INT32:
+    case FieldDescriptor::CPPTYPE_INT64:
+    case FieldDescriptor::CPPTYPE_UINT32:
+    case FieldDescriptor::CPPTYPE_UINT64:
+    case FieldDescriptor::CPPTYPE_FLOAT:
+    case FieldDescriptor::CPPTYPE_DOUBLE:
+    case FieldDescriptor::CPPTYPE_BOOL:
       return true;
-    case internal::WireFormatLite::CPPTYPE_STRING:
+    case FieldDescriptor::CPPTYPE_STRING:
       return false;
     default:
       return false;

